### PR TITLE
[Semi-Modular] SEVA Fixes

### PR DESCRIPTION
**Pull request by gonenoculer5**
## About The Pull Request
Migrates the SEVA Suit from /obj/item/clothing/suit/hooded/explorer/seva to /obj/item/clothing/suit/hooded/seva and make appropriate migrations of code from explorer to ensure it inherits the correct temperature flags and such, to prevent inconsistancies.
## How This Contributes To The Skyrat Roleplay Experience
N/A
## Changelog

:cl: Gonenoculer5
balance: SEVA suits no longer accept goliath plates, as was intended by the original PR author, and now add two seconds of stun to goliath grabs. Dont shoot the messenger.
fix: A recent factory mixup in the core worlds made it so that SEVA suits were comming with plate hooks for enviromental suits. Centcom has corrected this, and as such SEVA suits are now back to their manufactuing spec.
code: Relocated SEVA from explorer subtype to its own, to avoid inheriting plate addition from the hooded/explorer/XYZ subtype.
/:cl:

### CODE REVIEW

**Comment by gonenoculer5** [comment]
(This was tested thoroughly on a private server, it is appropriately fireproof, does not burn, you can walk in lava for a second and get a full firestack to test. Standing in a storm results in no damage as intended. Woo.)

**Comment by Iamgoofball** [comment]
:-1:

Remove the SEVA suit and make it a cosmetic subtype of the default mining outfit.

**Comment by gonenoculer5** [comment]
> 👎
> 
> Remove the SEVA suit and make it a cosmetic subtype of the default mining outfit.

It serves a genuine purpose though. I see no reason to remove it.

**Comment by GoldenAlpharex** [comment]
Lol, this isn't big enough to be a refactor.

**Review by GoldenAlpharex — APPROVED** [review]
So whoever ported it didn't add the code for the added stun duration for this suit?

Other than that, the code looks alright.

**Comment by GoldenAlpharex** [comment]
Also, update from master before we merge, so it runs the GBP action.

**Comment by gonenoculer5** [comment]
> So whoever ported it didn't add the code for the added stun duration for this suit?
> 
> Other than that, the code looks alright.

Yeah because it was ported from cit(?), and the original porter did not carry over the trait. According to cold it was intended to have a trait added called goliath_stun or something that would extend the stun duration, as it does on citbase, but it was never implemented. I could come back at a later date and add it, maybe. I'm not super familiar with trait code but supposedly its easy.

**Comment by gonenoculer5** [comment]
> Also, update from master before we merge, so it runs the GBP action.

This was made off my personal repo that had just been created that day, but then again I think cobalt fixed the GBP that day too so I probably have to update my master anyway.

**Comment by gonenoculer5** [comment]
> Lol, this isn't big enough to be a refactor.

Also whoops my bad. Yes this was intended to be a code improvement not a refactor!

**Comment by gonenoculer5** [comment]
> Also, update from master before we merge, so it runs the GBP action.

Done!

**Comment by GoldenAlpharex** [comment]
You didn't push you dork ;)

> > Also, update from master before we merge, so it runs the GBP action.
> 
> Done!

**Comment by SpaceVampire** [comment]
I feel like you should add the whole easily grabbed by goliath trait as a part of this PR, if you are already tweaking it.
If only because I doubt it is really going to get done otherwise, but that is just my two cents.

**Comment by gonenoculer5** [comment]
> I feel like you should add the whole easily grabbed by goliath trait as a part of this PR, if you are already tweaking it. If only because I doubt it is really going to get done otherwise, but that is just my two cents.

Unfortunately its more of a time thing than an interest thing, I would much rather push this fix/improvement now and return to it when I have time for the rest. Cold said the goliath stun time extention wasnt essential to it and it could always be added later.

**Comment by GoldenAlpharex** [comment]
> > I feel like you should add the whole easily grabbed by goliath trait as a part of this PR, if you are already tweaking it. If only because I doubt it is really going to get done otherwise, but that is just my two cents.
> 
> Unfortunately its more of a time thing than an interest thing, I would much rather push this fix/improvement now and return to it when I have time for the rest. Cold said the goliath stun time extention wasnt essential to it and it could always be added later.

To be fair, it's not critical or time-sensitive, so even if you take a week to do it, it would make it better than just having it be done by someone at a later date and everyone shooing them for it.

**Comment by gonenoculer5** [comment]
> > > I feel like you should add the whole easily grabbed by goliath trait as a part of this PR, if you are already tweaking it. If only because I doubt it is really going to get done otherwise, but that is just my two cents.
> > 
> > 
> > Unfortunately its more of a time thing than an interest thing, I would much rather push this fix/improvement now and return to it when I have time for the rest. Cold said the goliath stun time extention wasnt essential to it and it could always be added later.
> 
> To be fair, it's not critical or time-sensitive, so even if you take a week to do it, it would make it better than just having it be done by someone at a later date and everyone shooing them for it.

Eh just go ahead and merge it. I have to take time to poke and look at traits and ask azarak a few questions since he made the suggestion.

**Comment by gonenoculer5** [comment]
No longer fully modular because it has to edit code for goliaths.

**Comment by gonenoculer5** [comment]
@GoldenAlpharex So is this set to go? Its all ready and been tested, if someone wants to adjust the timing from higher than what it is, feel free. Its set to 12 seconds right now afaik.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm
Comment (GoldenAlpharex) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]: These numbers should be defines.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Just move those numbers to be defines and this should be good to go. Sorry for taking so long.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm
Comment (GoldenAlpharex) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]:
```suggestion
#define GOLIATH_STUN_TIME_SEVA 12 SECONDS //SKYRAT EDIT - SEVA Fix
```

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm
Comment (GoldenAlpharex) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]:
```suggestion
		 //SKYRAT EDIT START - GOLIATH STUN TIME FOR SEVA
		if(HAS_TRAIT(L,TRAIT_GOLIATH_STUN))
			L.Stun(GOLIATH_STUN_TIME_SEVA)
		else
			L.Stun(100)
		//SKYRAT EDIT END
```

**Review by GoldenAlpharex — APPROVED** [review]
Going to commit my suggestions and then probably merge this once CI passes.

**Comment by gonenoculer5** [comment]
pfft why is CI not passing now

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm
Comment (GoldenAlpharex) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]: There's mixed indentations here and that's what's messing up the linters. Check this in VSC and make sure it's all just tabs.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm
Comment (gonenoculer5) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]: It should be. I dont do anything BUT tab indentations. One moment.

Comment (gonenoculer5) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm]: Yeah. Everything in here is indented with tabs.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**
